public/js/gf-giftaid-field-frontend.js

```diff
@@ -1,57 +1,67 @@
-function gfGiftAidOnInputChange(elem) {
-  if (!(elem instanceof HTMLElement)) {
-    return;
-  }
+document.addEventListener('DOMContentLoaded', initGiftAid);
 
-  if (! elem.classList.contains('ginput_total') && ! elem.classList.contains('ginput_amount')) {
+/**
+ * Updates the gift aid display, based on a chosen field changing.
+ */
+function initGiftAid() {
+  const gravityForm = document.querySelector('.gform_wrapper');
+  if (!(gravityForm instanceof HTMLElement)) {
     return;
   }
-
-  const parentForm = elem.closest('form');
-  if (!(parentForm instanceof HTMLFormElement)) {
+  const totalSelector = totalFieldSelector(gravityForm, '.ginput_amount');
+  if (!totalSelector) {
     return;
   }
+  window.gform.addAction('gform_input_change', (elem) => {
+      const value = elem.value;
+      const valueFloat = parseFloat(value);
+      const giftAidFloat = valueFloat * 1.25;
+      const donation = valueFloat.toFixed(2);
+      const giftAidAmount = giftAidFloat.toFixed(2);
+      if (!donation || !giftAidAmount) {
+        return;
+      }
+      updateGiftAidDisplay(gravityForm, donation, giftAidAmount);
+    }, 10);
```

Comment:
```suggestion
  window.gform.addAction('gform_input_change', function (elem) {
      const donationValue = parseFloat(elem.value);
      const giftAidValue = donationValue * 1.25;
      const donationRounded = donationValue.toFixed(2);
      const giftAidRounded = giftAidValue.toFixed(2);
      if (!donationRounded || !giftAidRounded) {
        return;
      }
      updateGiftAidDisplay(gravityForm, donation, giftAidAmount);
    }, 10);
```
remember arrow function is not supported in all browsers yet and we are not transpiling with a build step.

Reply:
i changed variables/constants here as well btw in my suggestion @Barny-Thorpe